platform/openide.windows/src/org/openide/windows/ModeUtility.java

```diff
@@ -0,0 +1,104 @@
+ * <br>
+ * <br>
+ * The configuration of each Mode can be expressed in XML. This class gives access
+ * to that XML with a mechanism to reproduce a Mode from XML so you can construct 
+ * ways to save Mode and TopComponent combinations, which can subsequently be
+ * chosen and reloaded at a later time.
+ * <br>
+ * <br>
+ * Note that this is a different approach to how the Platform tends to save the layout
+ * of TopComponents at shut-down time in the Windows2Local file system. Note also that 
+ * this is not connected with the deprecated notion of {@link Workspace}.
+ * 
+ * @see http://wiki.apidesign.org/wiki/ExtendingInterfaces
+ * 
+ * @author Mark Phipps
+ */
+public final class ModeUtility {
+
+    private ModeUtility() {
+    }
+
+    /**
+     * Expose the Mode's configuration as XML.
+     * 
+     * @param mode the {@link Mode} whose XML configuration is required.
+     * @return the XML of the Mode's configuration or {@code null} if not supported.
+     */
+    public final static String toXml(Mode mode) {
+        return mode instanceof Mode.Xml
+                ? ((Mode.Xml) mode).toXml()
+                : null;
+    }
+    
+    /**
+     * Given some XML, attempts to ask the WindowManager to create a Mode that can
+     * subsequently be used to dock a TopComponent into. Usually this will be an 
+     * anonymous Mode.
+     * 
+     * @param windowManager typically WindowManager.getDefault()
+     * @param xml the XML that was originally produced by {@link #toXml}
+     */
+    public final static void createModeFromXml(WindowManager windowManager, String xml) {
+        if(windowManager instanceof WindowManager.ModeManager) {
+            ((WindowManager.ModeManager)windowManager).createModeFromXml(xml);
+        }
+    }
+    
+    /**
+     * Before restoring anonymous Modes, it is useful to update whatever defined Modes
+     * may exist like editor, explorer etc., so that all the Modes will eventually
+     * re-appear in the desired locations.
+     * 
+     * @param windowManager typically WindowManager.getDefault()
+     * @param xml the XML that was originally produced by {@link #toXml}
+     */
+    public final static void updateModeConstraintsFromXml(WindowManager windowManager, String xml) {
+        if(windowManager instanceof WindowManager.ModeManager) {
+            ((WindowManager.ModeManager)windowManager).updateModeConstraintsFromXml(xml);
+        }
+    }
+    
+    /**
+     * Before restoring a whole bunch of Modes (for example with XML that has been
+     * previously saved somewhere and now loaded), it is useful to remove the
+     * anonymous modes from the system.
+     * 
+     * @param windowManager typically WindowManager.getDefault()
+     * @param mode the {@link Mode} to remove
+     * @return success if the Mode can no longer be found in the Window Manager
+     */
+    public final static boolean removeMode(WindowManager windowManager, Mode mode) {
```

Comment:
What about `Mode.Xml.remove()`?

Reply:
That's an interface - not sure what you mean...

Reply:
Yeah, it is an interface, but a newly defined one. E.g. it is possible to put new methods into it, right?

Reply:
Sure, but at the point of removing a Mode from the WindowManager, there is no XML in play.
A Mode doesn't have to be an instance of Mode.XML to be removed. As far as I can see all Modes are ModeImpls anyway, which are defined with XML and the Mode.XML interface and ModeUtilities class provides the non-breaking-api way of accessing that.

Now I implemented the remove(), updateConstraints() and createMode() methods as abstract methods of WindowManager as you suggested.

Maybe you can judge these changes on my next push?